src/pages/patientView/mutationalSignatures/MutationalSignatureBarChart.tsx

```diff
@@ -0,0 +1,346 @@
+        );
+        const legendObjects = getLegendEntriesBarChart(labelObjects);
+        const lengthLegendObjects = getCenterPositionLabelEntries(
+            legendObjects
+        );
+        const legendOjbectsToAdd = createLegendLabelObjects(
+            lengthLegendObjects,
+            legendObjects,
+            labels
+        );
+        const centerOfBoxes = this.formatColorBoxLegend;
+        const xScale = getxScalePoint(labelObjects, 60, 1040);
+        const legendLabelsChart: JSX.Element[] = [];
+        legendOjbectsToAdd.forEach((item, i) => {
+            legendLabelsChart.push(
+                <VictoryLabel
+                    x={
+                        this.props.version != 'ID'
+                            ? centerOfBoxes[i].props.x +
+                              0.5 * centerOfBoxes[i].props.width
+                            : xScale(item.value)
+                    }
+                    y={8}
+                    width={this.props.width}
+                    text={item.group}
+                    style={{ fontSize: '13px', padding: 5 }}
+                    textAnchor={'middle'}
+                />
+            );
+        });
+        return legendLabelsChart;
+    }
+
+    @computed get formatColorBoxLegend() {
+        const groupedData = _.groupBy(
+            getColorsForSignatures(this.props.data),
+            'group'
+        );
+        const labels = Object.keys(groupedData);
+        const legendLabels = getColorsForSignatures(this.props.data).map(
+            entry => ({
+                group: entry.group,
+                label: entry.label,
+                color: entry.colorValue,
+                subcategory: entry.subcategory,
+            })
+        );
+        const xScale = getxScalePoint(legendLabels, 60, 1040);
+        const legendEntries = getLegendEntriesBarChart(legendLabels);
+        const lengthLegendObjects = getLengthLabelEntries(legendEntries);
+        const legendInfoBoxes = formatLegendObjectsForRectangles(
+            lengthLegendObjects,
+            legendEntries,
+            labels,
+            this.props.version
+        );
+        const legendLabelBoxes: JSX.Element[] = [];
+        legendInfoBoxes.forEach((item: DrawRectInfo) => {
+            legendLabelBoxes.push(
+                <rect
+                    x={xScale(item.start)}
+                    y={15}
+                    fill={item.color}
+                    width={
+                        xScale(item.end)! - xScale(item.start)! > 0
+                            ? xScale(item.end)! - xScale(item.start)!
+                            : 6
+                    }
+                    height="15px"
+                />
+            );
+        });
+        return legendLabelBoxes;
+    }
+
+    @computed get getSubLabelsLegend() {
+        const groupedData = _.groupBy(
+            getColorsForSignatures(this.props.data),
+            'group'
+        );
+        const labels = Object.keys(groupedData);
```

Comment:
better name

Reply:
I have created this computed because the same constants were created in `formatLegendTopaxisPoints` and `formatColorBoxLegend`:
```
@computed get getMutationalSignaturesGroupLabels():string[]{
        return Object.keys(this.getGroupedData)
    }
```